cmd/util/ledger/migrations/atree_register_migration.go

```diff
@@ -144,9 +161,60 @@ func (m *AtreeRegisterMigrator) MigrateAccount(
 		})
 	}
 
+	// Check storage health after migration, if enabled.
+	if m.checkStorageHealthAfterMigration {
+		mr, err := newMigratorRuntime(address, newPayloads)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create migrator runtime: %w", err)
+		}
+
+		err = checkStorageHealth(address, mr.Storage, newPayloads)
+		if err != nil {
+			m.log.Warn().
+				Err(err).
+				Str("account", address.Hex()).
+				Msg("storage health check after migration failed")
+		}
+	}
+
 	return newPayloads, nil
 }
 
+func checkStorageHealth(
```

Comment:
Once this PR is on master, when we get it into the Cadence 1.0 feature branch, we can refactor the existing code in the Cadence 1.0 migration to just use this function instead 👌

Reply:
I moved this function and `domains` to migrations/utils.go since they are reusable (commit 6ab8300125afbdf45d6d275d8e7628da9c2f0bb9).

Reply:
Great, thank you!